app/src/main/java/com/example/greehousecontroller/data/repository/CO2Repository.java

```diff
@@ -93,14 +108,21 @@ public void setThreshold(String greenhouseId, Threshold newThreshold){
             @Override
             public void onResponse(Call<Threshold> call, Response<Threshold> response) {
                 if (response.isSuccessful()){
-                    Log.i("Api-co2-st", response.body().toString());
-                    threshold.setValue(response.body());
+                    if(response.body() != null){
+                        Log.i("Api-co2-st", response.body().toString());
+                        threshold.setValue(response.body());
+                    }
                 }
+
+                if(!response.isSuccessful()){
+                        Toast.makeText(app.getApplicationContext(), R.string.unable_to_access_server, Toast.LENGTH_SHORT);
```

Comment:
Minor comment, instead of unable_to_access_server it could be unable_to_update_threshold